### DISCUSSION TITLE
Reset API state from specific endpoint

### DISCUSSION THREAD

**Discussion by iakovosvo** [reduxjs/redux-toolkit#2582] [Q&A] [answered]
I know there is a `resetApiState` util which resets the whole state of an API. Wondering if there is a way to reset the state from a specific endpoint and not the whole API state?

**Comment by phryneas** [comment]
You could just add a specific tag to that and invalidate it.

**Reply by iakovosvo** [reply]
Can I use `invalidateTags` for a query? Tried to export it from `util` and dispatch it with the tag type but didn't seen to work.

**Comment by phryneas** [comment, marked as answer]
That's what it is there for.

**Reply by iakovosvo** [reply]
This seems not remove all queries from the cache though. I was jus wondering if there is a trivial way to completely empty all queries cache (while using tags). For example I have a search bar UI and a list of elements and I want to clear the list when I clear the search input. Is there some way to do that without storing the result locally with `useState`?

**Reply by phryneas** [reply]
I think you are just asking the wrong question - this is about the hook, not about the cache state value.
Have you tried using `currentData` instead of `data` from the hook?

**Reply by iakovosvo** [reply]
I am using `currentData` & `isFetching` but I have also tried with `data` without any success. Not sure if this is the expected behaviour but when I empty the input value & run `invalidateTags` all queries from that specific tag type are removed from the cache, except from the last one, which is the one I want removed since I want to empty the list. So I just wanted to know if there is a recommended way of doing this (I just want to empty the list when the input value is completely empty). With the old Redux I could have done something like making a `CLEAR_LIST` action, which would just return the existing state with the items list being an empty array.

**Reply by phryneas** [reply]
So you called
```js
useListQuery("")
```
and it gave you an empty list. (1)

Then you called
```js
useListQuery("someSearchTerm")
```
and it gave you a list with results. (2)

Now you clear the search box.
Your code will call
```js
useListQuery("")
```
again.
And it will give you that same empty list (1) again.

Am I understanding something wrong right now?

**Reply by iakovosvo** [reply]
I am actually using the lazy query. Setting the term to to an empty string is what I am doing now in order to achieve what I want but that also makes a request with an empty term, which I was hoping to avoid.

**Reply by phryneas** [reply]
You could also just use the normal `query` hook with `skipToken` as an argument when you don't want the request to be made, that would give you an `undefined` result.

**Reply by iakovosvo** [reply]
That could actually work. Main reason I went with the lazy query was because I wanted to debounce the callback but I guess I can make a useDebounceTerm instead. Thanks for taking the time to respond to all my questions!
